### PR TITLE
use get_option instead of directly internal dict

### PR DESCRIPTION
**Pull request by bcoca**
##### SUMMARY
as get_option can populate it, while direct access cannot

##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

callbacks/default
##### ANSIBLE VERSION

```
2.7
```

### CODE REVIEW

File: lib/ansible/plugins/callback/default.py [hunk 1]
```diff
@@ -54,6 +54,15 @@ def __init__(self):
         self._last_task_name = None
         super(CallbackModule, self).__init__()
 
+    def get_option(key, default=None):
```
Comment (agaffney) [inline lib/ansible/plugins/callback/default.py]: This should have a `self` param

File: lib/ansible/plugins/callback/default.py [hunk 2]
```diff
@@ -54,6 +54,15 @@ def __init__(self):
...
+        # use 'non standard' get_option as other plugins derive from this one and might not document options correctly
+        option = default
+        try:
+            option = super(CallbackModule, self).get_option(key)
+        except (AttributeError, KeyError):
+            pass
+        return option
+
```
Comment (agaffney) [inline lib/ansible/plugins/callback/default.py]: Why not just add this logic to the base class's `get_option()` function?

Comment (bcoca) [reply lib/ansible/plugins/callback/default.py]: because you should not need a default, this is a special case for 3rd party derived modules from default

Comment (agaffney) [reply lib/ansible/plugins/callback/default.py]: Sure, but this assumes that there is nowhere else that this kind of thing comes up. I've heard of people doing their own private connection plugin that inherits from `ssh` and modifies some behaviors. With as many options that are specified by the config fragment in the `ssh` connection plugin's docs, this same situation could easily crop up.

Comment (bcoca) [reply lib/ansible/plugins/callback/default.py]: we already handle that, but mostly on try/except instead

File: lib/ansible/plugins/callback/default.py [hunk 3]
```diff
@@ -304,7 +316,7 @@ def v2_playbook_on_stats(self, stats):
         self._display.display("", screen_only=True)
 
         # print custom stats
-        if self._plugin_options.get('show_custom_stats', DEFAULT_SHOW_CUSTOM_STATS) and stats.custom:
+        if self.show_custom_stats:
```
Comment (agaffney) [inline lib/ansible/plugins/callback/default.py]: Is the `and stats.custom` part of the condition not actually needed?

Comment (bcoca) [reply lib/ansible/plugins/callback/default.py]: nope, that was overzealous deletion

**Review by agaffney — APPROVED** [review]

**Pull request merged**